src/main/java/org/spongepowered/api/data/persistence/DataStore.java

```diff
@@ -115,18 +117,36 @@ default DataView serialize(DataManipulator dataManipulator) {
      * Provides a {@link DataStore} for a single {@link Key}.
      * <p>
      *     Note that default deserializers do not support {@link Collection}, {@link Map} or Array types!
-     *     Use {@link Builder#key(Key, BiConsumer, Function)} for these.
+     *     Use {@link Builder.SerializersStep#key(Key, BiConsumer, Function)} for these.
      * </p>
      *
      * @param key The data key
      * @param dataQuery The dataQuery to serialize this key under
-     * @param typeToken The dataHolder type
+     * @param typeTokens The dataHolder types
      *
      * @return The new data store
      */
     @SafeVarargs
-    static <T> DataStore of(Key<Value<T>> key, DataQuery dataQuery, TypeToken<? extends DataHolder>... typeToken) {
-        return builder().key(key, dataQuery).holder(typeToken).build();
+    static <T> DataStore of(Key<Value<T>> key, DataQuery dataQuery, TypeToken<? extends DataHolder>... typeTokens) {
+        return builder().pluginData(key.getKey()).holder(typeTokens).key(key, dataQuery).build();
+    }
+
+    /**
+     * Provides a {@link DataStore} for a single {@link Key}.
+     * <p>
+     *     Note that default deserializers do not support {@link Collection}, {@link Map} or Array types!
+     *     Use {@link Builder.SerializersStep#key(Key, BiConsumer, Function)} for these.
+     * </p>
```

Comment:
Much like the comment above, may want to mention that the varargs of types is optional to specify what types the datastore will support?